quickwit/quickwit-indexing/src/split_store/local_split_store.rs

```diff
@@ -381,17 +382,11 @@ impl LocalSplitStore {
         } else {
             return Ok(None);
         };
-        let split_file_res = split_store_lock.move_out(split_ulid, output_dir_path).await;
-        match split_file_res {
-            Ok(Some(split_path)) => Ok(Some(split_path)),
-            Ok(None) => Ok(None),
-            Err(storage_err) if storage_err.kind() == StorageErrorKind::DoesNotExist => {
-                warn!(split_id = split_id, error = ?storage_err, "Cached split file/folder is missing.");
-                split_store_lock.split_folders.remove(&split_ulid);
```

Comment:
I think the only case where we execute this line, is on a code path where we already removed the element from `split_folders` and found `None`.